packages/entities/entities-plugins/src/components/PluginCustomGrid.vue

```diff
@@ -0,0 +1,274 @@
+  config: {
+    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    required: true,
+    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+      if (!config || !['konnect', 'kongManager'].includes(config?.app)) return false
+      if (!config.getCreateRoute) return false
+      return true
+    },
+  },
+  /**
+   * Whether or not user has rights to create custom plugins
+   */
+  canCreateCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Whether or not user has rights to delete custom plugins
+   */
+  canDeleteCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Whether or not user has rights to edit custom plugins
+   */
+  canEditCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Plugins to display in the grid
+   */
+  pluginList: {
+    type: Object as PropType<PluginCardList>,
+    default: () => ({}),
+  },
+  /**
+   * @param {boolean} noRouteChange if true, let consuming component handle event when clicking on a plugin
+   * Used in conjunction with `@plugin-clicked` event
+   */
+  noRouteChange: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Number of plugins to always have visible (never will be collapsed)
+   */
+  pluginsPerRow: {
+    type: Number,
+    default: 4,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'plugin-clicked', plugin: PluginType): void,
+  (e: 'revalidate'): void,
+}>()
+
+const { i18n: { t } } = composables.useI18n()
+const shouldCollapsedCustomPlugins = ref(true)
+
+const emitPluginData = (plugin: PluginType) => {
+  emit('plugin-clicked', plugin)
+}
+
+const modifiedCustomPlugins = computed((): PluginType[] => {
+  if (props.config.app === 'kongManager') {
+    return []
+  }
+
+  const customPlugins: PluginType[] = JSON.parse(JSON.stringify(props.pluginList))[PluginGroup.CUSTOM_PLUGINS] || []
+
+  // ADD CUSTOM_PLUGIN_CREATE as the first card if allowed creation
+  return props.canCreateCustom && !props.noRouteChange && props.config.createCustomRoute
+    ? [{
+      id: 'custom-plugin-create',
+      name: t('plugins.select.tabs.custom.create.name'),
+      available: true,
+      group: PluginGroup.CUSTOM_PLUGINS,
+      description: t('plugins.select.tabs.custom.create.description'),
+    } as PluginType].concat(customPlugins)
+    : customPlugins
+})
+
+const getPluginCards = (type: 'all' | 'visible' | 'hidden') => {
+  if (type === 'all') {
+    return modifiedCustomPlugins.value
+  } else if (type === 'visible') {
+    return modifiedCustomPlugins.value.slice(0, props.pluginsPerRow)
+  }
+
+  return modifiedCustomPlugins.value.slice(props.pluginsPerRow)
+}
+
+// text for plugin group "view x more" label
+const triggerLabel = computed((): string => {
+  const totalCount = getPluginCards('all')?.length || 0
+  const hiddenCount = getPluginCards('hidden')?.length || 0
```

Comment:
question: it looks like `getPluginCards` always returns an array, right?

If yes, I'm not sure you need the fallbacks to `0` here?